### DISCUSSION TITLE
Why is redirect not acting as nextjs redirect

### DISCUSSION THREAD

**Discussion by ScreamZ** [amannn/next-intl#1415] [General]
Hey @amannn 

Just try this code within an async function

```ts
  const a: null | string = (() => (Math.random() > 0 ? "hello" : null))();

  if (!a) {
    redirect("/"); // Should be enough to infer
  }

  console.log(a)
``` 

Why using redirect from next-intl, a is infered `string|null`  at console.log level and only `string`  when using nextjs redirect function ?

Both are returning `never`

**Comment by amannn** [comment]
It's unfortunately a limitation in TypeScript: https://github.com/amannn/next-intl/issues/823

**Reply by ScreamZ** [reply]
Oh i see :'( I'll try some workaround listed, but I understand the limitation THanks
